packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -577,32 +577,14 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
     }
     assert(_manager == null || child != _manager.rootScope, "Reparenting the root node isn't allowed.");
     assert(!ancestors.contains(child), 'The supplied child is already an ancestor of this node. Loops are not allowed.');
-    FocusNode oldPrimaryFocus;
```

Comment:
It isn't needed, because I realized that the reason there was a problem here was that the setAsFocusedChild was happening as a side effect of setting the focus, and that was all I actually wanted to do as part of reparenting.  Setting the focus "for reals" here caused problems because rebuilding the widget tree involved reparenting things, and that shouldn't affect which thing had focus, it should just maintain the integrity of the focused child information.

There was a test, but it was just testing that the focused child info still had integrity, which it is still doing, but not that the focus changed in this way, so nothing needed to be deleted.